src/types.ts

```diff
@@ -182,10 +181,43 @@ export interface ProviderOptions<T = unknown> {
   contextTransformer?: ContextTransformer<T>;
 }
 
+export enum StandardResolutionReasons {
+  /**
+   * Indicates that the feature flag is targeting
+   * 100% of the targeting audience,
+   * e.g. 100% rollout percentage
+   */
+  TARGETING_MATCH = 'TARGETING_MATCH',
+  /**
+   * Indicates that the feature flag is targeting
+   * a subset of the targeting audience,
+   * e.g. less than 100% rollout percentage
+   */
+  SPLIT = 'SPLIT',
+  /**
+   * Indicates that the feature flag is disabled
+   */
+  DISABLED = 'DISABLED',
+  /**
+   * Indicates that the default value for
+   * the feature flag is returned
+   */
+  DEFAULT = 'DEFAULT',
+  /**
+   * Indicates an unknown issue occurred during evaluation
+   */
+  UNKNOWN = 'UNKNON',
+  /**
+   * Indicates that an error occurred during evaluation
+   */
+  ERROR = 'ERROR',
+}
+export type ResolutionReason = StandardResolutionReasons | string;
```

Comment:
TypeScript will treat this as a string. That means you won't get auto completion in the IDE. That's not a dealbreaker but perhaps something like this would work better.

```typescript
export type ResolutionReason = keyof typeof StandardResolutionReasons | (string & Record<never, never>)
```

This isn't prefect either since it's using the key and not the value. It also may be a bit more difficult to understand what it's doing. It's up to you.

Reply:
Cool, didn't even think about that. I have updated it